s3transfer/manager.py

```diff
@@ -79,6 +80,16 @@ def __init__(self,
             are already retried by botocore (this default is 5). The
             ``num_download_attempts`` does not take into account the
             number of exceptions retried by botocore.
+
+        :param max_in_memory_upload_chunks: The number of chunks that can
+            be stored in memory at a time for all ongoing uploads. This
+            pertains to chunks of data that need to be stored in memory
+            during an upload if the data is sourced from a file-like object.
+            The total expected memory footprint due to in-memory upload
+            chunks is roughly equal to:
+
+                max_in_memory_upload_chunks * multipart_chunksize
```

Comment:
Looking at the code again, it actually appears to be that the calculation to determine how many in-flight upload chunks should be ``max_in_memory_upload_chunks * multipart_chunksize + submission_concurrency * multipart_chunksize`` because even though the submission gets blocked by the executor, the submission thread will still call the next iteration and that in-memory chunk will be blocked in getting submitted to request executor. Does this make sense?

I wonder if there is a better abstraction I can do to handle this. The hard part is that I need to communicate the number of in-flight chunks between two thread pools especially since you cannot make the assumption that a specific type of input will produce in-flight chunks (you need to know if it is multipart or not). I wonder if I can just do a better job of naming the tuning parameter and updating the description?